test_files/class/class.ts

```diff
@@ -1,38 +1,107 @@
+// This test exercises the various ways classes and interfaces can interact.
+// There are three types of classy things:
+//   interface, class, abstract class
+// And there are two keywords for relating them:
+//   extends, implements
+// You can legally use them in almost any configuration the cross product implies;
+// for example, you can "implements" a class though it's more rare than the
+// other options.
+
+// Three declarations, one for each type of thing.
 interface Interface {
   interfaceFunc(): void;
 }
-
-class Super {
-  superFunc(): void {}
+class Class {
+  classFunc(): void {}
+}
+abstract class AbstractClass {
+  abstract abstractFunc(): void;
+  nonAbstractFunc(): void { }
 }
 
+// Write out all permutations:
+// 1) interface implements
+// 2) interface extends
+// 3) class implements
+// 4) class extends
 
-class Implements implements Interface, Super {
-  interfaceFunc(): void {}
-  superFunc(): void {}
+// 1) interface implements.
+// No examples; this is not legal TypeScript.
```

Comment:
I don't understand what this comment refers to. Should this have a number from above?